### PR TITLE
promisified redis commands

### PR DESCRIPTION
**Pull request by zalun**

### CODE REVIEW

**Comment by zalun** [comment]
Do not merge - there is something strange in tests - [SyntaxError: Unexpected token o]

File: engine/index.js [hunk 1]
```diff
@@ -303,48 +303,49 @@ const statusFields = ['firefox_status', 'spec_status', 'opera_status',
...
+          }
+        });
+      }
     });
-  })).then((client) => {
-    client.quit();
-    return features;
-  }).catch((err, client) => {
+  })
+  .catch((err) => {
```
Comment (marco-c) [inline engine/index.js]: Can you move the catch at the end?

File: engine/index.js [hunk 2]
```diff
@@ -303,48 +303,49 @@ const statusFields = ['firefox_status', 'spec_status', 'opera_status',
...
-      if (err) {
-        console.error('ERROR: ' + err);
-        reject(err, client);
-      }
+  .then((redisClient) => {
+    client = redisClient;
+  })
+  .then(() => redis.get(client, 'status'))
+  .then((oldStatus) => {
+    if (oldStatus) {
```
Comment (marco-c) [inline engine/index.js]: Huge nit: I preferred the old:
```JavaScript
try {
  oldStatus = JSON.parse(oldStatus);
} catch (e) {
  console.error(e, oldStatus);
}
if (!oldStatus) {
  oldStatus = {};
}
```

File: engine/index.js [hunk 3]
```diff
@@ -303,48 +303,49 @@ const statusFields = ['firefox_status', 'spec_status', 'opera_status',
...
+  let client;
   return redis.getClient(dbTestNumber)
-  .then((client) => new Promise((resolve, reject) => {
-    client.get('status', (err, oldStatus) => {
-      if (err) {
-        console.error('ERROR: ' + err);
-        reject(err, client);
-      }
+  .then((redisClient) => {
+    client = redisClient;
```
Comment (marco-c) [inline engine/index.js]: Nit: Here and in the following functions, if you wrap the rest inside this function you don't need to define a `client` variable.

File: engine/index.js [hunk 4]
```diff
@@ -371,29 +373,21 @@ function saveData(features, dbTestNumber) {
...
-        resolve(client);
-      });
-    });
-  }))
-  .catch((err, client) => {
+      return redis.hmset(client, 'changelog', date, JSON.stringify(changedData));
+    }
+    console.log('DEBUG: no changes found');
+  })
+  .catch((err) => {
```
Comment (marco-c) [inline engine/index.js]: Can you move catch at the end?

File: engine/index.js [hunk 5]
```diff
@@ -303,48 +303,49 @@ const statusFields = ['firefox_status', 'spec_status', 'opera_status',
...
-  })).then((client) => {
-    client.quit();
-    return features;
-  }).catch((err, client) => {
+  })
+  .catch((err) => {
     console.error('ERROR:', err);
+  })
+  .then(() => {
     client.quit();
```
Comment (marco-c) [inline engine/index.js]: Should we wait for quit to finish?

**Comment by digitarald** [comment]
@zalun can we unblock this?

File: engine/index.js (hunk 5 above)
Comment (zalun) [inline engine/index.js]: it happens immediately and we do not open it straight after

**Pull request merged**